packages/synapse-react-client/vite.config.ts

```diff
@@ -1,48 +1,25 @@
-import { NodeModulesPolyfillPlugin } from '@esbuild-plugins/node-modules-polyfill'
-import react from '@vitejs/plugin-react'
 import { resolve } from 'path'
-import { defineConfig } from 'vite'
-import svgr from 'vite-plugin-svgr'
-import jotaiDebugLabel from 'jotai/babel/plugin-debug-label'
-import jotaiReactRefresh from 'jotai/babel/plugin-react-refresh'
+import { mergeConfig } from 'vite'
+import viteConfig from 'vite-config'
+import { externalizeDeps } from 'vite-plugin-externalize-deps'
+import dts from 'vite-plugin-dts'
 
-export default defineConfig({
-  root: './src',
+export default mergeConfig(viteConfig, {
+  root: '.',
+  build: {
+    emptyOutDir: false,
+    outDir: './dist',
+    lib: {
+      entry: resolve(__dirname, 'src/index.ts'),
+      name: 'SRC',
+      fileName: 'index',
+      formats: ['es', 'cjs'],
+    },
+  },
   plugins: [
-    react({ babel: { plugins: [jotaiDebugLabel, jotaiReactRefresh] } }),
-    svgr({
-      svgrOptions: {
-        plugins: ['@svgr/plugin-jsx'],
-        ref: true,
-      },
+    externalizeDeps(),
```

Comment:
Do not bundle any dependencies, since the consumer will be able to dynamically load them with their bundler + package manager